extensions/amp-story/1.0/amp-story-tooltip.js

```diff
@@ -202,46 +239,71 @@ export class AmpStoryTooltip {
   }
 
   /**
-   * Builds tooltip and attaches it depending on the clicked element content and
+   * Builds tooltip and attaches it depending on the target's content and
    * position.
-   * @param {!Element} clickedEl
+   * @param {!Element} target
    * @private
    */
-  attachTooltipToEl_(clickedEl) {
-    const elUrl = clickedEl.getAttribute('href');
-    if (!isProtocolValid(elUrl)) {
-      user().error(TAG, 'The tooltip url is invalid');
-      return;
-    }
+  updateTooltipEl_(target) {
+    this.updateTooltipText_(target);
+    this.updateTooltipIcon_(target);
+    this.positionTooltip_(target);
+  }
 
-    const iconUrl = clickedEl.getAttribute('data-tooltip-icon');
-    if (!isProtocolValid(iconUrl)) {
-      user().error(TAG, 'The tooltip icon url is invalid');
-      return;
+  /**
+   * Updates tooltip behavior depending on the target.
+   * @param {!Element} target
+   */
+  updateTooltipBehavior_(target) {
+    if (matches(target, tooltipDelegatableSelectors().LINK)) {
+      addAttributesToElement(dev().assertElement(this.tooltip_),
+          dict({'href': this.getElementHref_(target)}));
     }
+    else if (matches(target, tooltipDelegatableSelectors().TWITTER)) {
```

Comment:
For the link test above, it makes sense to check the CSS selector since you want to verify that it's a link and has a HREF.
But here, we don't need that CSS check, and I don't think we want to write a new `else if` every time we add a component, this should scale automatically as we add new `EXPANDABLE_COMPONENTS`.

`if (Object.values(EXPANDABLE_COMPONENTS).includes(target.tagName.toLowerCase()))`

Nit: indent

Reply:
Done.